src/lldp/lldppkt.rs

```diff
@@ -0,0 +1,1023 @@
+        let system_name: String = String::from_utf8(
+            buf.get_vec(buf.remaining())
+                .ok_or(pktparser::ParseError::UnexpectedEndOfInput)?,
+        )
+        .map_err(|e| pktparser::ParseError::InvalidArgument(e.to_string()))?;
+
+        Ok(SystemName(system_name))
+    }
+}
+
+impl Serialise for SystemName {
+    fn to_wire(&self) -> std::result::Result<Vec<u8>, std::io::Error> {
+        Ok(self.0.as_bytes().to_vec())
+    }
+}
+
+impl std::fmt::Display for SystemName {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        write!(f, "SystemName: {}", self.0)
+    }
+}
+
+#[derive(Debug)]
+pub struct SystemDescription(String);
+
+impl Deserialise for SystemDescription {
+    fn from_wire(
+        buf: &mut pktparser::Buffer<'_>,
+    ) -> std::result::Result<Self, pktparser::ParseError> {
+        let system_description: String = String::from_utf8(
+            buf.get_vec(buf.remaining())
+                .ok_or(pktparser::ParseError::UnexpectedEndOfInput)?,
+        )
+        .map_err(|e| pktparser::ParseError::InvalidArgument(e.to_string()))?;
+
+        Ok(SystemDescription(system_description))
+    }
+}
+
+impl Serialise for SystemDescription {
+    fn to_wire(&self) -> std::result::Result<Vec<u8>, std::io::Error> {
+        Ok(self.0.as_bytes().to_vec())
+    }
+}
+
+impl std::fmt::Display for SystemDescription {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        write!(f, "SystemDescription: {}", self.0)
+    }
+}
+
+#[derive(Debug)]
+pub struct SystemCapabilities {
+    pub sys_cap: u16,
+    pub enabled_cap: u16,
+}
+
+impl Deserialise for SystemCapabilities {
+    fn from_wire(
+        buf: &mut pktparser::Buffer<'_>,
+    ) -> std::result::Result<Self, pktparser::ParseError> {
+        if buf.remaining() != 4 {
+            return Err(pktparser::ParseError::InvalidArgument(format!(
+                "SystemCapabilities TLV length must be 4 but got {}",
+                buf.remaining(),
+            )));
+        }
+
+        let sys_cap = buf
+            .get_be16()
+            .ok_or(pktparser::ParseError::UnexpectedEndOfInput)?;
+        let enabled_cap = buf
+            .get_be16()
+            .ok_or(pktparser::ParseError::UnexpectedEndOfInput)?;
+
+        Ok(SystemCapabilities {
+            sys_cap,
+            enabled_cap,
+        })
+    }
+}
+
+impl Serialise for SystemCapabilities {
+    fn to_wire(&self) -> std::result::Result<Vec<u8>, std::io::Error> {
+        let mut result = Vec::new();
+        WriteBytesExt::write_u16::<BigEndian>(&mut result, self.sys_cap)?;
+        WriteBytesExt::write_u16::<BigEndian>(&mut result, self.enabled_cap)?;
+        Ok(result)
+    }
+}
+
+impl std::fmt::Display for SystemCapabilities {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        write!(
+            f,
+            "SystemCapabilities system: {}, enabled: {}",
```

Comment:
these should probably be hex {:x} or it should enumerate what the capabilities are by name.